api/envoy/config/core/v3/base.proto

```diff
@@ -313,6 +313,24 @@ message HeaderMap {
   repeated HeaderValue headers = 1;
 }
 
+// A path that is watched for changes, e.g. by inotify on Linux.
+message WatchedPath {
```

Comment:
This looks much nicer!

Would it make sense to split this up into two separate messages, `WatchedDirectory` and `WatchedFile`, so that the user doesn't need to specify the event type?  That might be a more ergonomic way of expressing the same thing.

Reply:
Switched back to WatchedDirectory, I think that's all we need per @tsaarni.